### PR TITLE
Cinematic JPG/PNG image support

### PR DESCRIPTION
**Pull request by 0lvin**
Add support of JPG/PNG format and small cleanup in soft render.

### CODE REVIEW

File: src/client/refresh/gl1/gl1_draw.c [hunk 1]
```diff
@@ -330,7 +330,7 @@ RDraw_FadeScreen(void)
 }
 
 void
-RDraw_StretchRaw(int x, int y, int w, int h, int cols, int rows, byte *data)
+RDraw_StretchRaw(int x, int y, int w, int h, int cols, int rows, const byte *data, int bits)
```
Comment (Yamagi) [inline src/client/refresh/gl1/gl1_draw.c:333]: Declaring 'data' as const leads to a new warning further down below:

```
src/client/refresh/gl1/gl1_draw.c: In function 'RDraw_StretchRaw':
src/client/refresh/gl1/gl1_draw.c:451:40: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  451 |                                 source = data + cols * row;
      |                                        ^
src/client/refresh/gl1/gl1_draw.c:482:32: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  482 |                         source = data + cols * row;
      |
```

File: src/client/refresh/soft/sw_draw.c [hunk 2]
```diff
@@ -321,15 +321,20 @@ RE_Draw_StretchRaw
 =============
 */
 void
-RE_Draw_StretchRaw (int x, int y, int w, int h, int cols, int rows, byte *data)
+RE_Draw_StretchRaw (int x, int y, int w, int h, int cols, int rows, const byte *data, int bits)
```
Comment (Yamagi) [inline src/client/refresh/soft/sw_draw.c:324]: The same as above:

```
src/client/refresh/soft/sw_draw.c: In function 'RE_Draw_StretchRaw':
src/client/refresh/soft/sw_draw.c:360:30: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  360 |                 image_scaled = data;
      |
```

**Review by Yamagi — COMMENTED** [review]
I have added two small inline comments regarding new warnings. Beside them this looks okay.

File: src/client/refresh/gl1/gl1_draw.c (hunk 1 above)
Comment (0lvin) [reply src/client/refresh/gl1/gl1_draw.c:333]: Thank you! Fixed!

File: src/client/refresh/soft/sw_draw.c (hunk 2 above)
Comment (0lvin) [reply src/client/refresh/soft/sw_draw.c:324]: Thank you! Fixed.

**Comment by 0lvin** [comment]
I have added one small fix, I have found that Heretic 2 models/sprites has bigger limit for count of sprites. Could you please look?

**Comment by Yamagi** [comment]
I'm sorry for my long delays. I don't have much time to spend for Q2 right now. It should turn better in a few weeks... Anyways, this now looks good. Thank you for bearing with me. :)

**Pull request merged**